### ISSUE TITLE
[device-port] [axon7] ZTE Axon 7

### ISSUE THREAD

**Issue by TuxThePenguin0** [Halium/projectmanagement#131]
Tree: halium-7.1

- [x] Create manifest (has been pulled into repo)
- [x] Boot image and system image build successfully
- [x] Device boots into rootfs, `usb: Manufacturer: GNU/Linux Device` appears in `dmesg` on host.
- [x] LXC container starts and does not crash (Only in UT rootfs)
- [ ] libhybris tests
  * [x] test_gps (Only in UT rootfs)
  * [ ] test_hwcomposer
  * [ ] test_lights
  * [x] test_vibrator
  * [x] test_wifi (Only in UT rootfs)
  * [ ] test_sensors
  * [ ] test_audio
  * [ ] test_camera
  * [x] test_input (Only in UT rootfs)
  * [ ] test_nfc
  * [ ] test_recorder

**Information:**
Due to being based on LOS 14 flashing the phone back to the LOS 14 modem, bootloader and partition layout is required.

I have removed the NFC sources from the manifest as (at least without modification) they do not compile. Due to this it is unlikely that this device will ever have NFC working in any Halium based distributions although to my knowledge this is not an issue as no distributions either support or make use of NFC anyway.

**First boot!**
![IMG_20190712_195704](https://user-images.githubusercontent.com/33564505/61120364-2e6c7480-a4e0-11e9-994c-ce338c319b96.jpg)

**Whats working on UBports Xenial armhf (Halium boot):**
- [x] USB
- [x] Display
- [x] Touch
- [x] Hardware keys
- [x] SD card slot
- [ ] Sound
- [x] WiFi (partial, requires terminal command `echo sta > /sys/module/wlan/parameters/fwpath` to enable)
- [ ] Bluetooth
- [ ] Modem
- [x] Sensors
- [ ] Camera
- [x] Flashlight
- [ ] Fingerprint sensor
- [ ] NFC (likely never)

**Comment by TuxThePenguin0** [comment]
Oh actually I forgot that I did modify the kernel configuration, once I find the time I will change the configuration to use my own kernel repo, please do not accept this pull request yet

**Comment by TuxThePenguin0** [comment]
Ok I have updated the kernel configuration in my own repo and modified the manifest to use my kernel repo, it *should* be good to go now

**Comment by TuxThePenguin0** [comment]
I'll make a pull request to add the fixup mountpoints for the Axon 7 soon

**Comment by TuxThePenguin0** [comment]
Don't worry this port isn't abandoned or anything, it may even boot, I don't know, I have just been away from home for the last few days, and unable to do any work on it, I will continue work in a day or two

**Comment by TuxThePenguin0** [comment]
I am unsure how to solve this kernel panic issue, I have tried to contact people on the Halium Telegraph server but have gotten no response yet
https://gist.github.com/TuxThePenguin0/3a549c031a7b168888b13bfea84232d4
Any help would be greatly appreciated

**Comment by TuxThePenguin0** [comment]
Sorry for the relatively frequent pull requests, this device is a very tricky port due to the stock Lineage OS kernel for it not booting at all for unknown reasons

**Comment by TuxThePenguin0** [comment]
These are two more pull requests for things that I probably should have done at the start
https://github.com/Halium/hybris-boot/pull/76
https://github.com/Halium/halium-devices/pull/154

**Comment by TuxThePenguin0** [comment]
First major milestone! I got the device to boot into recovery, however this does not get me much further since I had to use an unmodified kernel to get it to boot at all

**Comment by TuxThePenguin0** [comment]
The device is booting into recovery!

**Comment by TuxThePenguin0** [comment]
I am having some trouble getting the LXC containers running but some other people who are way smarter than me are having similar issues on different devices so we are trying to find a fix

**Comment by TuxThePenguin0** [comment]
Currently the LXC container is working on the Ubuntu Touch rootfs however graphics is not working yet and neither are most of the hardware tests

**Comment by joelkahan** [comment]
Any updates?

**Comment by TuxThePenguin0** [comment]
Yeah sorry I haven't been working on this port lately since I have had other work and my other phone (Nexus 5X) gave up the ghost a second time, not to mention that I was having some trouble getting things like the sensors to work.

I have since bought another phone which I have spent some time getting UT working on and I will get back to the Axon 7 port eventually, it will just take me some time.

In the mean time any help is greatly appreciated :)

**Comment by niansa** [comment]
@TuxThePenguin0 hi, any progress? Would be very nice to share source code and some build in case you've given this up. :-)
I might continue this

**Comment by TuxThePenguin0** [comment]
If anyone wants to take over this they can, I have little motivation to work on UT ports at the moment and my Axon 7 bricked itself so there's no reason for me to continue working on the port

**Comment by GeoEduca** [comment]
Greetings, I am trying to start a job based on this model: zte nubia z5s mini **nx406e**
I just hope to find information to start.
I will try to be active on these 2 sites. For their develpment.
https://www.reddit.com/r/LineageOS/comments/r48d61/contribute_with_a_new_model_zte_nubia_z5s_mini/
https://forums.ubports.com/topic/7057/a-query-about-the-device-zte-nubia-z5s-mini-nx406e
any help is appreciated
